diff --git a/dlls/winedmo/libavcodec/pcm_byte_order_reverse_bsf.c b/dlls/winedmo/libavcodec/pcm_byte_order_reverse_bsf.c
index 2237b1eaa84..00fc983e2b5 100644
--- a/dlls/winedmo/libavcodec/pcm_byte_order_reverse_bsf.c
+++ b/dlls/winedmo/libavcodec/pcm_byte_order_reverse_bsf.c
@@ -28,15 +28,15 @@
 
 #define IS_EMPTY(pkt) (!(pkt)->data && !(pkt)->side_data_elems)
 
-struct AVBSFInternal {
+typedef struct PCMByteOrderReverseBSFContext {
     AVPacket *buffer_pkt;
     int eof;
-};
+} PCMByteOrderReverseBSFContext;
 
 /* From FFmpeg */
 int ff_bsf_get_packet(AVBSFContext *ctx, AVPacket **pkt)
 {
-    AVBSFInternal *bsfi = ctx->internal;
+    PCMByteOrderReverseBSFContext *bsfi = ctx->priv_data;
     AVPacket *tmp_pkt;
 
     if (bsfi->eof)
@@ -77,7 +77,7 @@ static enum AVCodecID reverse_codec_id(enum AVCodecID codec_id)
 
 static int init(AVBSFContext *ctx)
 {
-    if (ctx->par_in->channels <= 0 || ctx->par_in->sample_rate <= 0)
+    if (ctx->par_in->ch_layout.nb_channels <= 0 || ctx->par_in->sample_rate <= 0)
         return AVERROR(EINVAL);
     if (ctx->par_in->bits_per_coded_sample % 8u)
         return AVERROR(EINVAL);
