diff --git a/server/request.c b/server/request.c
index 2691e0c..df1aeea 100644
--- a/server/request.c
+++ b/server/request.c
@@ -648,7 +648,7 @@ static char *create_server_dir( int force )
     if (asprintf( &base_dir, "%s/.wineserver", config_dir ) == -1)
         fatal_error( "out of memory\n" );
 #else
-    if (asprintf( &base_dir, "/tmp/.wine-%u", getuid() ) == -1)
+    if (asprintf( &base_dir, "/data/data/com.gebox.emu/files/usr/tmp/.wine" ) == -1)
         fatal_error( "out of memory\n" );
 #endif
     create_dir( base_dir, &st2 );
diff -uNr a/dlls/crypt32/unixlib.c b/dlls/crypt32/unixlib.c
--- a/dlls/crypt32/unixlib.c
+++ b/dlls/crypt32/unixlib.c
@@ -619,6 +619,7 @@
  "/usr/local/share/certs/",
  "/etc/sfw/openssl/certs",
  "/etc/security/cacerts",  /* Android */
+ "/data/data/com.gebox.emu/files/usr/glibc/etc/ca-certificates/cacert.pem"
 };
 
 static void load_root_certs(void)
diff -uNr a/programs/winemenubuilder/winemenubuilder.c b/programs/winemenubuilder/winemenubuilder.c
--- a/programs/winemenubuilder/winemenubuilder.c
+++ b/programs/winemenubuilder/winemenubuilder.c
@@ -1708,7 +1708,7 @@
     if (_wgetenv( L"XDG_DATA_DIRS" ))
        dirs = xwcsdup( *wgetenv( L"XDG_DATA_DIRS" ));
     else
-        dirs = xwcsdup( L"/usr/local/share/:/usr/share/" );
+        dirs = xwcsdup( L"/data/data/com.gebox.emu/files/usr/glibc/local/share:/data/data/com.gebox.emu/files/usr/glibc/share" );
 
     ret = add_mimes(xdg_data_dir, mime_types);
     if (ret)
diff -uNr a/dlls/ntdll/unix/server.c b/dlls/ntdll/unix/server.c
--- a/dlls/ntdll/unix/server.c
+++ b/dlls/ntdll/unix/server.c
@@ -1233,7 +1233,7 @@
 #ifdef **ANDROID**  /* there's no /tmp dir on Android */
     asprintf( &dir, "%s/.wineserver/server-%llx-%llx", config_dir, (unsigned long long)dev, (unsigned long long)ino );
 #else
-    asprintf( &dir, "/tmp/.wine-%u/server-%llx-%llx", getuid(), (unsigned long long)dev, (unsigned long long)ino );
+    asprintf( &dir, "/data/data/com.gebox.emu/files/usr/tmp/.wine/server-%llx-%llx", (unsigned long long)dev, (unsigned long long)ino );
 #endif
 
     return dir;
